crates/ty_python_semantic/src/types/display.rs

```diff
@@ -120,13 +183,27 @@ impl Display for DisplayRepresentation<'_> {
                         .expect("Specialization::tuple() should always return `Some()` for `KnownClass::Tuple`")
                         .display_with(self.db, self.settings)
                         .fmt(f),
-                    (ClassType::NonGeneric(class), _) => f.write_str(class.name(self.db)),
+                    (ClassType::NonGeneric(class), _) => {
+                        if self.settings.qualified {
+                            f.write_fmt(format_args!("{}.{}", self.class_parents(class).join("."), class.name(self.db)))
+                        } else {
+                            f.write_str(class.name(self.db))
+                        }
```

Comment:
It seems like we do this in several places -- we should define a function for it.